lib/internal/streams/async_iterator.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const { promisify } = require('util');
+
+class Item {
+  constructor(value, done) {
```

Comment:
Or at least, do away with the `done` slot for the vast majority of objects.

Reply:
why not just use plain object with 2 fields? Why everything has to be a class?

Reply:
@YurySolovyov not everything has to be a class, but in this particular case there are several benefits in naming the objects:

 - When looking at heap allocation profiles, it makes them easier to recognize (rather than just Object).
 - When debugging and looking at stack traces, you get more useful information since the objects are named.

As a platform, this makes naming objects appealing. I'm not sure it's worth it but it's definitely a reasonable call. I would name it `AsyncIteratorRecord` to be ore similar to the spec.

Reply:
This shouldn't have a big perf impact either way. I looked at perf of object / class instance creation in my work on `nextTick` and it's negligible.

I don't like the idea of reusing it as that means the returned object can no longer be stored.

Reply:
I would also add that it make sure that objects created in multiple places maintain the same shape, this helps V8 in the optimization process.

Reply:
> do away with the done slot for the vast majority of objects.

We can't do that if we want to conform to the spec for what `next` should return.

Reply:
The one suggestion I might have here is making `done = false` so we don't have to repeat that everywhere. It's the default state after all...

Reply:
@apapirovski I meant - we can put `done = false` on the prototype but I suspect it'll be slow (and might not conform).

I want this to eventually be fast to be useful :)